scripts/batched-llava-caption.py

```diff
@@ -0,0 +1,344 @@
+# Copyright 2022 MosaicML Diffusion authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Script to LLaVA caption an image dataset."""
+
+import os
+import threading
+import time
+from argparse import ArgumentParser, Namespace
+from typing import Optional
+
+import torch
+import wandb
+from composer.utils import dist
+from huggingface_hub import snapshot_download
+from torchvision import transforms
+from tqdm.auto import tqdm
+
+try:
+    from llava.constants import DEFAULT_IMAGE_TOKEN  # type: ignore
+    from llava.constants import DEFAULT_IM_END_TOKEN, DEFAULT_IM_START_TOKEN, IMAGE_TOKEN_INDEX  # type: ignore
+    from llava.conversation import SeparatorStyle, conv_templates  # type: ignore
+    from llava.mm_utils import get_model_name_from_path, tokenizer_image_token  # type: ignore
+    from llava.model.builder import load_pretrained_model  # type: ignore
+    from llava.utils import disable_torch_init  # type: ignore
+except ImportError:
+    raise ImportError(
+        'LLaVA is not installed. Please install it with `pip install llava@git+https://github.com/haotian-liu/LLaVA.git`'
+    )
```

Comment:
Use `raise from` statement here to explicitly chain the exceptions

Reply:
Done!